conjurapi/client.go

```diff
@@ -186,22 +175,26 @@ func NewClientFromEnvironment(config Config) (*Client, error) {
 		return NewClientFromKey(config, *loginPair)
 	}
 
-	loginPair, err = LoginPairFromNetRC(config)
-	if err == nil && loginPair.Login != "" && loginPair.APIKey != "" {
-		return NewClientFromKey(config, *loginPair)
-	}
+	if storageProvider, _ := createStorageProvider(config); storageProvider != nil {
```

Comment:
> Having trouble wrapping my head around the lifecycle of the storageProvider created here. Isn't it initialized and attached to the client in `NewClientFromOidcCode` -> `newClientWithAuthenticator` -> `NewClient`?

Yeah this is confusing. In `NewClientFromEnvironment()` we need to see if there are saved credentials we can use. I'm going to move it into a separate function.

> In general, `NewClientFromEnvironment` is getting a bit unwieldy. Maybe not avoidable since it is sort of a catch-all but it would be nice to break out some of the JWT and OIDC logic at some point

Agreed. This should really be cleaned up and also made easier to test.

Reply:
@adamouamani I'm thinking we should have a quality store (1-2 points) to clean up this function and add unit tests. Thoughts?